### PR TITLE
chore: add secure password to root user in mysql CI db

### PR DESCRIPTION
**Pull request by diegomarquezp**
This is intended to fix a vulnerability brought by [b/296915631](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/b/296915631). The fix is about [SQL_WEAK_ROOT_PASSWORD](http://vulnstore/#advisories&expand=true&q=publisher:SHA+reference:SQL_WEAK_ROOT_PASSWORD)

### CODE REVIEW

File: spring-cloud-gcp-samples/pom.xml [hunk 1]
```diff
@@ -166,7 +166,7 @@
 								<gcs-local-directory>/tmp/gcp_integration_tests/integration_storage_sample</gcs-local-directory>
 								<spring.cloud.gcp.sql.instance-connection-name>spring-cloud-gcp-ci:us-central1:testmysql</spring.cloud.gcp.sql.instance-connection-name>
 								<spring.cloud.gcp.sql.database-name>code_samples_test_db</spring.cloud.gcp.sql.database-name>
-								<spring.datasource.password>test</spring.datasource.password>
+								<spring.datasource.password>${spring-cloud-gcp-ci-mysql-db-root-password}</spring.datasource.password>
 								<spring.r2dbc.password>test</spring.r2dbc.password>
```
Comment (meltsufin) [inline spring-cloud-gcp-samples/pom.xml:170]: Do we have to set the password at all? AFAIK we're using integration authentication which is safe without a db password. Also, wouldn't we have to fix the r2dbc password as well?

Comment (diegomarquezp) [reply spring-cloud-gcp-samples/pom.xml:170]: This is intended to fix a vulnerability brought by [b/296915631](b/296915631). 
I'm currently testing this config to determine where we need to add this password

Comment (diegomarquezp) [reply spring-cloud-gcp-samples/pom.xml:170]: Added this to PR description as well

Comment (diegomarquezp) [reply spring-cloud-gcp-samples/pom.xml]: I converted this to a draft in the meantime

**Comment by suztomo** [comment]
In PR description, can you paste the error message with stacktrace this pull request fixes?

**Comment by diegomarquezp** [comment]
```
Caused by: org.hibernate.HibernateException: Unable to determine Dialect without JDBC metadata (please set 'javax.persistence.jdbc.url', 'hibernate.connection.url', or 'hibernate.dialect')
	at org.hibernate.engine.jdbc.dialect.internal.DialectFactoryImpl.determineDialect(DialectFactoryImpl.java:188)
	at org.hibernate.engine.jdbc.dialect.internal.DialectFactoryImpl.buildDialect(DialectFactoryImpl.java:87)
	at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.initiateService(JdbcEnvironmentInitiator.java:274)
	at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.initiateService(JdbcEnvironmentInitiator.java:34)
	at org.hibernate.boot.registry.internal.StandardServiceRegistryImpl.initiateService(StandardServiceRegistryImpl.java:119)
	at org.hibernate.service.internal.AbstractServiceRegistryImpl.createService(AbstractServiceRegistryImpl.java:264)
	... 110 more

[INFO] 
[INFO] Results:
[INFO] 
Error:  Errors: 
Error:    RegistrationKotlinSampleApplicationIntegrationTests.testRegisterPersonSuccessful » IllegalState Failed to load ApplicationContext for [WebMergedContextConfiguration@761b9314 testClass = com.example.RegistrationKotlinSampleApplicationIntegrationTests, locations = [], classes = [com.example.app.Application], contextInitializerClasses = [], activeProfiles = [], propertySourceLocations = [], propertySourceProperties = ["spring.datasource.***", "spring.cloud.gcp.sql.instance-connection-name=spring-cloud-gcp-ci:us-central1:testmysql", "spring.cloud.gcp.sql.database-name=code_samples_test_db", "spring.test.mockmvc.print=none", "org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true", "server.port=0"], contextCustomizers = [[ImportsContextCustomizer@20e9c165 key = [org.springframework.boot.test.autoconfigure.web.servlet.MockMvcAutoConfiguration, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcWebClientAutoConfiguration, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcWebDriverAutoConfiguration, org.springframework.boot.autoconfigure.security.oauth2.client.servlet.OAuth2ClientAutoConfiguration, org.springframework.boot.autoconfigure.security.oauth2.resource.servlet.OAuth2ResourceServerAutoConfiguration, org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration, org.springframework.boot.autoconfigure.security.servlet.SecurityFilterAutoConfiguration, org.springframework.boot.autoconfigure.security.servlet.UserDetailsServiceAutoConfiguration, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcSecurityConfiguration, org.springframework.boot.test.autoconfigure.web.reactive.WebTestClientAutoConfiguration]], org.springframework.boot.test.context.filter.ExcludeFilterContextCustomizer@4983159f, org.springframework.boot.test.json.DuplicateJsonObjectContextCustomizerFactory$DuplicateJsonObjectContextCustomizer@44040454, org.springframework.boot.test.mock.mockito.MockitoContextCustomizer@0, org.springframework.boot.test.web.client.TestRestTemplateContextCustomizer@784c3487, org.springframework.boot.test.autoconfigure.actuate.observability.ObservabilityContextCustomizerFactory$DisableObservabilityContextCustomizer@1f, org.springframework.boot.test.autoconfigure.properties.PropertyMappingContextCustomizer@4b3fa0b3, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverContextCustomizerFactory$Customizer@119020fb, org.springframework.boot.test.context.SpringBootTestAnnotation@90ca17ca], resourceBasePath = "src/main/webapp", contextLoader = org.springframework.boot.test.context.SpringBootContextLoader, parent = null]
```

File: spring-cloud-gcp-samples/pom.xml [hunk 2]
```diff
@@ -166,7 +166,7 @@
 								<gcs-local-directory>/tmp/gcp_integration_tests/integration_storage_sample</gcs-local-directory>
 								<spring.cloud.gcp.sql.instance-connection-name>spring-cloud-gcp-ci:us-central1:testmysql</spring.cloud.gcp.sql.instance-connection-name>
 								<spring.cloud.gcp.sql.database-name>code_samples_test_db</spring.cloud.gcp.sql.database-name>
-								<spring.datasource.password>test</spring.datasource.password>
+								<spring.datasource.password>${db.root.password}</spring.datasource.password>
```
Comment (meltsufin) [inline spring-cloud-gcp-samples/pom.xml:169]: If you define `--define "spring.datasource.password=${DB_PASSWORD}"`, do you still need this line at all?

**Comment by suztomo** [comment]
29 successful checks

**Comment by suztomo** [comment]
"Accessing data with MySQL" document https://spring.io/guides/gs/accessing-data-mysql/ has `spring.datasource.password`. (I was expecting spring.mysql.password or something like that) Good.

File: spring-cloud-gcp-samples/pom.xml (hunk 2 above)
Comment (diegomarquezp) [reply spring-cloud-gcp-samples/pom.xml]: Ah, true it would not be necessary. I just changed it

**Pull request merged**